src/develop/develop.c

```diff
@@ -2681,6 +2688,13 @@ void dt_dev_zoom_move(dt_dev_viewport_t *port,
   dt_pthread_mutex_unlock(&dev->history_mutex);
   dt_pthread_mutex_unlock(&(darktable.control->global_mutex));
 
+  if( old_zoom == port->zoom
+      && old_closeup == port->closeup
+      && fabsf(old_zoom_x - port->zoom_x) < 1e-6
```

Comment:
I think the overall idea is **very** good and it avoids some pipe recalculations. Nevertheless i have some jitter-around while mouse-scroll zooming in.

As both zoom_x & zoom_y are coordinates it think the "eps" of 1e-6 is by far too small, i tested with 2.0f and everything is rock-stable

Reply:
Not sure I notice much difference personally. I don't know what you mean by "jitter" but I have noticed that there now seems to be some sort of "progressive update" when zooming out, which wasn't there before -- if I zoom out two "clicks" of my zoom wheel then the display updates the roi of the first zoom click and then the second, where before it updated the whole image at once. It's a bit distracting but I'm not sure whether it's intentional or not.

Reply:
To reproduce the "jitter" use mouse scroll in darkroom being already at 100% and try to increase zoom. You will see - in master and with this pr - the darkroom image slightly moving around for a pixel or even two, this is likely due to some rounding and roi corrections.

As any dragging around for just one pixel - either by a mouse drag or caused by rounding - doesn't make sense to me we could avoid reprocessing/exposing in such a case. Maybe we can find a better threshold than 2.0 but i think the idea is really good as it avoids getting data via pixelpipe - even if that cost is close to zero because of caching.

About the drawing-whole-image-at-once, the new behaviour is intentional.

Reply:
See #15688, also solved via this pr for me.

Reply:
> As both zoom_x & zoom_y are coordinates it think the "eps" of 1e-6 is by far too small

You are right of course!

> i tested with 2.0f and everything is rock-stable

The risk is that you may end up 1 pixel from the edge of the crop and not be able to move to the _real_ end. So you could end up with a whole row of contrasting pixels that you meant to crop out and _checked_ (by moving "all the way" to the edge _while zoomed in_) and it still showing up in your final image. So I think .5 pixel would be the maximum. But since distortions can stretch 1 pixel in the RAW to multiple pixels in the processed image, maybe 0.1? I'll submit that so we all can test.

> I don't know what you mean by "jitter"

Me neither. To me, jitter is repeated jumping back-and-forth between two states. Not meandering around or slowly creeping. But people seem to use it with slightly different meanings which doesn't help the clarity of reported issues.

> there now seems to be some sort of "progressive update" when zooming out, which wasn't there before

Is "before" 4.4.2 or before this PR? The progressive update was always there, but in 4.4.2 it was just not shown; multiple (many) recalcs could go on in the background, but the scaled up preview image continued to be shown until you stopped moving/zooming. Now you get to see the intermediate/partial results, so there is a quicker response.

I had not been able to reproduce #15688 (tbh I was not sure I understood it) or #15747 yet (which has a very helpful video and might be a duplicate of #15688), but I hopefully added a complete fix for both of those as well, so I'm optimistically going to mark this PR as "fixes" them both. @axelg-de would you please test if this PR solves your issue?

Reply:
> To me, jitter is repeated jumping back-and-forth between two states

At least in the real-time audio processing or timings jitter is also associated with "in phase" ...

What i meant was, that the image is not constantly at the same position while further zooming in at the same mouse position. Likely due to `dt_dev_distort_backtransform_locked()`.

Anyway, tested you latest commit and it all looks good to me now.

Reply:
> @AxelG-DE would you please test if this PR solves your issue?

I did test and also documented in the [aforementioned issue 15747](https://github.com/darktable-org/darktable/issues/15747#issuecomment-1826826775). It does improve a lot, but still the image jumps upon release the left mouse button. Find video in the issue...